astropy/io/ascii/core.py

```diff
@@ -1016,26 +1016,44 @@ class BaseOutputter:
     """Output table as a dict of column objects keyed on column name.  The
     table data are stored as plain python lists within the column objects.
     """
+    # User-defined converters which gets set in ascii.ui if a `converter` kwarg
+    # is supplied.
     converters = {}
+
     # Derived classes must define default_converters and __call__
 
     @staticmethod
     def _validate_and_copy(col, converters):
         """Validate the format for the type converters and then copy those
         which are valid converters for this column (i.e. converter type is
         a subclass of col.type)"""
+        # Allow specifying a single converter instead of a list of converters.
+        # The input `converters` must be a ``type`` value that can init np.dtype.
+        try:
+            # Don't allow list-like things that dtype accepts
+            assert type(converters) is type
```

Comment:
The result of calling `type(converters)` (on the left) is the class of `converters`, but `type` (on the right) is a built-in function. Doesn't this line always result in an `AssertionError`?

Reply:
I agree this looked a bit funny when I was writing it. The proof is in the pudding below, but calling the built-in `type` function on a class returns `type`. Basically the class of a class is `type`.
```
In [2]: type(np.int64) is type
Out[2]: True

In [3]: type(['hello']) is type
Out[3]: False

In [4]: type(1) is type
Out[4]: False

In [5]: type(int) is type
Out[5]: True
```

Reply:
The reason I was confused about this was because I thought that `type` is a built-in function, but actually it is a class. My conclusion that the line in question would always result in an `AssertionError` therefore followed from a false assumption.

Reply:
maybe add a one-liner comment in the code, so this would not trick another person reading it

Reply:
> Basically the class of a class is `type`.

This isn't true for classes with a meta class, including the most common case "ABCMeta".

```python
from abc import ABCMeta

class Cls(metaclass = ABCMeta):
    pass

type(Cls) is type
```

Reply:
``issubclass(Cls, type)`` fails as well...

Reply:
But ``isinstance(Cls, type)`` works!